driver/src/price_finding/price_finder_interface.rs

```diff
@@ -20,6 +22,38 @@ impl Default for Fee {
     }
 }
 
+#[derive(Clone, Copy, PartialEq)]
+pub enum OptimizationModel {
+    NAIVE,
+    MIP,
+    NLP,
+}
+
+impl From<String> for OptimizationModel {
+    fn from(optimization_model_string: String) -> OptimizationModel {
+        match optimization_model_string.to_lowercase().as_str() {
+            "mip" => OptimizationModel::MIP,
+            "nlp" => OptimizationModel::NLP,
+            // the naive solver is the standard solver.
+            _ => OptimizationModel::NAIVE,
+        }
+    }
+}
+
+impl OptimizationModel {
+    pub fn to_args(self) -> Result<&'static str, PriceFindingError> {
```

Comment:
nit: naming should be try_to_args since it returns a Result. I'm wondering if it would even be fine to panic here and not return an error, since this would be an invalid logical state if we end up with the naive option here.